File: archive/d/dart/factorial.dart
// Issue 4969
void main(List<String> args){
  const String error_message = "Usage: please input a non-negative integer";
  try{

    int factorial = 1;
    int num_needed = int.parse(args[0]);
    if (num_needed.isNegative == true){
      print(error_message);
    }
    else{
      for (int i_index = 1; i_index <= num_needed; i_index ++){
        factorial *= i_index;
      }
      print(factorial.toString());
    }

  }
  catch(e){
    print(error_message);
  }
}

File: archive/s/scala/Factorial.scala
import scala.util.Try

object Factorial:
  private val usage = "Usage: please input a non-negative integer"

  def main(args: Array[String]): Unit =
    val result =
      args.headOption
        .flatMap(a => Try(a.toInt).toOption)
        .filter(_ >= 0)
        .map(factorial)

    result match
      case Some(value) => println(value)
      case None => println(usage)

  private def factorial(n: Int): Int =
    (1 to n).product